mkosi/__init__.py

```diff
@@ -7370,7 +7370,7 @@ def ensure_networkd(args: MkosiArgs) -> bool:
     return True
 
 
-def run_shell(args: MkosiArgs) -> None:
+def pre_run_shell(args: MkosiArgs) -> MkosiArgs:
```

Comment:
Instead of "pre_run_shell()", I'd name this function "run_shell_cmdline()" which is less generic and more clearly indicates that the function returns the cmdline for run_shell()

Also, wrong return type here, should be Sequence[str]